src/Models/UserModel.php

```diff
@@ -185,48 +192,107 @@ public function activate(User $user): void
     {
         $user->active = true;
 
-        $return = $this->save($user);
-
-        $this->checkQueryReturn($return);
+        $this->save($user);
     }
 
     /**
-     * Override the BaseModel's `save()` method to allow
-     * updating of user email, password, or password_hash
-     * fields if they've been modified.
+     * @param User $data
```

Comment:
I think we should still support arrays, and the callback events would just ignore them.

Reply:
Frankly, I am not sure why it is necessary to accept arrays, but I implemented it that way to try it out.